### ISSUE TITLE
New cop that detects operations with constant results

### ISSUE THREAD

**Issue by fatkodima** [rubocop/rubocop#11191]
Inspired by https://hexdocs.pm/credo/Credo.Check.Warning.OperationWithConstantResult.html

It should check for the code like
```ruby
x * 1
x * 0
x / 1
x ** 0
x ** 1
x % 1
```

This is almost always a mistake (or just left after debugging, probably) or just redundant code.
